src/plugins/dashboard/common/saved_dashboard_references.test.ts

```diff
@@ -41,28 +51,28 @@ describe('extractReferences', () => {
       },
       references: [],
     };
-    const updatedDoc = extractReferences(doc);
+    const updatedDoc = extractReferences(doc, deps);
 
     expect(updatedDoc).toMatchInlineSnapshot(`
-Object {
-  "attributes": Object {
-    "foo": true,
-    "panelsJSON": "[{\\"title\\":\\"Title 1\\",\\"panelRefName\\":\\"panel_0\\"},{\\"title\\":\\"Title 2\\",\\"panelRefName\\":\\"panel_1\\"}]",
-  },
-  "references": Array [
-    Object {
-      "id": "1",
-      "name": "panel_0",
-      "type": "visualization",
-    },
-    Object {
-      "id": "2",
-      "name": "panel_1",
-      "type": "visualization",
-    },
-  ],
-}
-`);
+      Object {
+        "attributes": Object {
+          "foo": true,
+          "panelsJSON": "[{\\"embeddableConfig\\":{},\\"title\\":\\"Title 1\\",\\"panelRefName\\":\\"panel_0\\"},{\\"embeddableConfig\\":{},\\"title\\":\\"Title 2\\",\\"panelRefName\\":\\"panel_1\\"}]",
```

Comment:
In each of these snapshots `embeddableConfig:{}` is added because now underlying code converts to embeddablePanel and back and we never clean up `embeddableConfig` when we convert from `EmbeddablePanel` to `dashboardSavedPanel`. I think having `embeddableConfig` is technically more correct and initial doc is just in not complete shape.

Other option could be to remove `embeddableConfig` in `convertSavedDashboardPanelToPanelState` if there is no keys.